From bee06aa0f13b6c2d35a5da84857a8e1c8c5b9b0e Mon Sep 17 00:00:00 2001
From: Dash Santosh <santdas36@gmail.com>
Date: Wed, 5 Jun 2024 13:32:12 +0530
Subject: [PATCH] add send packet logs

---
 libavcodec/decode.c | 18 ++++++++++++++++++
 1 file changed, 18 insertions(+)

diff --git a/libavcodec/decode.c b/libavcodec/decode.c
index 139cb6ecfa..80a59fc824 100644
--- a/libavcodec/decode.c
+++ b/libavcodec/decode.c
@@ -723,20 +723,35 @@ int attribute_align_arg avcodec_send_packet(AVCodecContext *avctx, const AVPacke
     int ret;
 
     if (!avcodec_is_open(avctx) || !av_codec_is_decoder(avctx->codec))
+    {
+        av_log(avctx, AV_LOG_ERROR, "init einval: %d\n", AVERROR(EINVAL));
         return AVERROR(EINVAL);
+    }
 
     if (dc->draining_started)
+    {
+        av_log(avctx, AV_LOG_ERROR, "draining: %d\n", AVERROR(EAGAIN));
         return AVERROR_EOF;
+    }
 
     if (avpkt && !avpkt->size && avpkt->data)
+    {
+        av_log(avctx, AV_LOG_ERROR, "einval failed: %d\n", AVERROR(EINVAL));
         return AVERROR(EINVAL);
+    }
 
     if (avpkt && (avpkt->data || avpkt->side_data_elems)) {
         if (!AVPACKET_IS_EMPTY(avci->buffer_pkt))
+        {
+            av_log(avctx, AV_LOG_ERROR, "av_packet_unref() failed: %d\n", ret);
             return AVERROR(EAGAIN);
+        }
         ret = av_packet_ref(avci->buffer_pkt, avpkt);
         if (ret < 0)
+        {
+            av_log(avctx, AV_LOG_ERROR, "av_packet_ref() failed: %d\n", ret);
             return ret;
+        }
     } else
         dc->draining_started = 1;
 
@@ -745,7 +760,10 @@ int attribute_align_arg avcodec_send_packet(AVCodecContext *avctx, const AVPacke
         av_log(avctx, AV_LOG_DEBUG, "avcodec_send_packet ret:%d\n", ret);
         av_log(avctx, AV_LOG_DEBUG, "avc condition: %d\n", ret < 0 && ret != AVERROR(EAGAIN) && ret != AVERROR_EOF);
         if (ret < 0 && ret != AVERROR(EAGAIN) && ret != AVERROR_EOF)
+        {
+            av_log(avctx, AV_LOG_ERROR, "decode_receive_frame_internal() failed: %d\n", ret);
             return ret;
+        }
     }
 
     return 0;
-- 
2.43.0.windows.1

